core/connection.js

```diff
@@ -619,3 +621,32 @@ Blockly.Connection.prototype.setShadowDom = function(shadow) {
 Blockly.Connection.prototype.getShadowDom = function() {
   return this.shadowDom_;
 };
+
+/**
+ * @return {string} A human-readable description of this Connection.
+ */
+Blockly.Connection.prototype.describe = function() {
+  var msg;
+  var block = this.sourceBlock_;
+  if (!block) {
+    return 'Orphan Connection';
+  } else if (block.outputConnection === this) {
```

Comment:
Why the ===?  At Google we only use === if we are explicitly distinguishing between null and undefined or somesuch.  Here and below.